crates/bevy_input/src/gamepad.rs

```diff
@@ -1,16 +1,272 @@
 //! The gamepad input functionality.
```

Comment:
One thing to consider with this new approach: by splitting this out from the generic `Axis` and `ButtonInput` resources, we're essentially committing to introducing special-cased logic for higher level input/action mapping APIs.

In their current forms, in theory you could build generic high level action logic that works with any `Axis` or `ButtonInput` resource.

Not a huge deal, but something to consider.

Reply:
I've generally had to special-case gamepads pretty hard in LWIM, due to their very unusual structure and the "what gamepad" problem. Users want to bind `GamepadButtonType`, but only `GamepadButton` can be stored in `ButtonInput`, which is tied to a specific gamepad. The same problem applies to axes :(

It might be nice to upstream a "this button on any gamepad" API that ties back into `ButtonInput`, but that doesn't need to and probably shouldn't be in this PR.

Reply:
Cool. I'm convinced that the special casing is possible and likely necessary. I'm down to move forward.